transaction.go

```diff
@@ -489,18 +514,62 @@ func (transaction *Transaction) ToBytes() ([]byte, error) {
 
 }
 
-func (transaction *Transaction) buildTransactions(untilIndex int) error {
-	for i := len(transaction.transactions); i < untilIndex; i++ {
-		data, err := protobuf.Marshal(transaction.signedTransactions[i])
+func (transaction *Transaction) signTransaction(index int) {
+	if len(transaction.signedTransactions[index].SigMap.SigPair) != 0 {
```

Comment:
should probably check if the index is outside bounds of the array here, I am not a fan of those out of bounds errors

Reply:
Yea, my reasoning here is that since this an internal API if we somehow provide an invalid index we shouldn't be checking that, but instead it should panic.